diff --git a/csrc/u8g2_message.c b/csrc/u8g2_message.c
index 15d3942..e9ed47d 100644
--- a/csrc/u8g2_message.c
+++ b/csrc/u8g2_message.c
@@ -35,7 +35,7 @@
 
 #include "u8g2.h"
 
-#define SPACE_BETWEEN_BUTTONS_IN_PIXEL 6
+#define SPACE_BETWEEN_BUTTONS_IN_PIXEL 12
 #define SPACE_BETWEEN_TEXT_AND_BUTTONS_IN_PIXEL 3
 
 uint8_t u8g2_draw_button_line(u8g2_t *u8g2, u8g2_uint_t y, u8g2_uint_t w, uint8_t cursor, const char *s)
diff --git a/csrc/u8g2_selection_list.c b/csrc/u8g2_selection_list.c
index 5d5736c..307705d 100644
--- a/csrc/u8g2_selection_list.c
+++ b/csrc/u8g2_selection_list.c
@@ -38,6 +38,8 @@
 #include "u8g2.h"
 
 #define MY_BORDER_SIZE 1
+#define MY_HEADER_CLEARANCE 6
+#define MY_BOX_EXPAND_X 4
 
 
 /*
@@ -53,6 +55,7 @@ void u8g2_DrawUTF8Line(u8g2_t *u8g2, u8g2_uint_t x, u8g2_uint_t y, u8g2_uint_t w
 {
   u8g2_uint_t d, str_width;
   u8g2_uint_t fx, fy, fw, fh;
+  u8g2_uint_t box_expand = 0;
 
   /* only horizontal strings are supported, so force this here */
   u8g2_SetFontDirection(u8g2, 0);
@@ -71,22 +74,24 @@ void u8g2_DrawUTF8Line(u8g2_t *u8g2, u8g2_uint_t x, u8g2_uint_t y, u8g2_uint_t w
     d -=str_width;
     d /= 2;
   }
-  else
-  {
+  else {
+    if (w == 0) {
+      box_expand = MY_BOX_EXPAND_X;
+    }
     w = str_width;
   }
 
   /* caluclate text box */
-  fx = x;
+  fx = x - box_expand;
   fy = y - u8g2_GetAscent(u8g2) ;
-  fw = w;
+  fw = w + box_expand * 2;
   fh = u8g2_GetAscent(u8g2) - u8g2_GetDescent(u8g2) ;
 
   /* draw the box, if inverted */
   u8g2_SetDrawColor(u8g2, 1);
   if ( is_invert )
   {
-    u8g2_DrawBox(u8g2, fx, fy, fw, fh);
+    u8g2_DrawRBox(u8g2, fx, fy, fw, fh, 2);
   }
 
   /* draw the frame */
@@ -96,7 +101,7 @@ void u8g2_DrawUTF8Line(u8g2_t *u8g2, u8g2_uint_t x, u8g2_uint_t y, u8g2_uint_t w
     fy--;
     fw +=2;
     fh +=2;
-    u8g2_DrawFrame(u8g2, fx, fy, fw, fh );
+    u8g2_DrawRFrame(u8g2, fx, fy, fw, fh, 2);
     border_size--;
   }
 
@@ -252,7 +257,7 @@ uint8_t u8g2_UserInterfaceSelectionList(u8g2_t *u8g2, const char *title, uint8_t
 		
 	  u8g2_DrawHLine(u8g2, 0, yy-line_height- u8g2_GetDescent(u8g2) + 1, u8g2_GetDisplayWidth(u8g2));
 		
-	  yy += 3;
+	  yy += MY_HEADER_CLEARANCE;
         }
         u8g2_DrawSelectionList(u8g2, &u8sl, yy, sl);
       } while( u8g2_NextPage(u8g2) );
